073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,150 @@
+* strimzi.io/cluster-ca-key-generation tracks the CA key generation to determine whether new end entity certificates are required
+* strimzi.io/cluster-ca-cert-generation tracks the CA certificate generation to determine whether components trust the CA
+
+## Motivation
+
+Removing the assumption that there is a _single_ root CA certificate stored in a special Kubernetes Secret that is directly used for trust would enable much more flexibility in CA handling.
+This will be compatible with commonly deployed certificate management solutions such as [Cert Manager][cmio], or [Vault][vault].
+These solutions generate certificates that (from the PoV of an end-entity certificate requester) are signed by arbitrary CAs.
+Using those solutions therefore requires TLS peers to be able to trust multiple root CA certificates.
+In addition these solutions cannot be responsible for rolling updates to components like Kafka.
+This means that end-entity certificate generation and the rollout of trust in the CA certificates referenced in those end-entity certificates, needs to be decoupled.
+This would be highly valuable for organizations with compliance requirements with regard to certificates.
+
+## Proposal
+
+Strimzi should be updated to decouple the management of the cluster and clients CA certificates, and the consumption of those CA certificates by components (such as ZooKeeper, Kafka and cluster operator).
+
+To do this we will introduce new Kubernetes Secrets that are used to keep track of CA certificates and their current state.
+These new Kubernetes Secrets are separate from the ones that are mounted into the components to use a truststore.
+The cluster operator will update these new Kubernetes Secrets when it renews cluster or client CA certificates.
+The cluster operator and user operator will copy the CA certificates from the new Kubernetes Secrets into the existing truststore Kubernetes Secrets that are mounted into components.
+The cluster operator and user operator will update the new Kubernetes Secrets to reflect whether the CA certificates are currently trusted and/or in use or not.
+This will make it easier to reason about which CA certificates are in use/trusted, and allow the management of CA certificates to be evolved further in future (see [Future changes](#future-changes)).
+These new Kubernetes Secrets can be updated by multiple processes (i.e the cluster operator when issuing the CA certificates, and the cluster and user operators when updating the state of CA certificates).
+
+### New Kubernetes Secret format for CA certificate state
+
+A new Kubernetes Secret is introduced which is primarily updated by the cluster operator in its role as a CA, but also updated by the cluster operator and user operator to reflect whether the individual CA certificates are trusted and/or in use in the cluster.
+
+```yaml
+kind: Secret
+metdata:
+  name: ${cluster}-${ca-type}-trust # e.g. foo-cluster-ca-trust
+type: strimzi.io/trust
```

Comment:
I think this would be the first time we've used/defined a Secret `type`. I think it would be good to define a json schema for any such types we define.